Cabal/Distribution/Types/TestSuite.hs

```diff
@@ -65,3 +66,11 @@ testModules test = (case testInterface test of
                      TestSuiteLibV09 _ m -> [m]
                      _                   -> [])
                 ++ otherModules (testBuildInfo test)
+
+-- | Get all the auto generated module names from a test suite.
+-- This are a subset of 'testModules'.
+testModulesAutogen :: TestSuite -> [ModuleName]
+testModulesAutogen test = (case testInterface test of
```

Comment:
This doesn't look right to me. In a library test suite, the declared module is generally written explicitly by the user. With this patch, you will stop including this module in the sdist. Bad!